awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -556,6 +563,23 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
         else:
             self.exit_json(**self.json_output)
 
+    # We need to be able to recursevly step through fields in the case of inputs for credentials.
+    # They are dicts and we can't just compare them at the top level because the dict returned from tower_cli will have fields like $encrypted$
```

Comment:
"from tower_cli" --> "from the API"?

Reply:
Also, this doesn't do $encrypted$ stuff, that's still a tricky issue that we can't fix without user consent.

It looks like this allows for no-op with weird falsy values from the API, and that makes sense.

Reply:
Comment fixed. And yes, we can't do anything with $encrypted$ so if you are using a parameter like `password` that comes through as $encrypted$ we have to assume that the object is changing. So we just let `new_input` compare to `$encrypted$`.